#593 add password requirements

martyav opened the pull request:
<!--
Check the Rancher docs issues to see if there is an existing issue for this pull request. If there is, enter the issue number below.
-->

Fixes #593

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, make sure to target the release branch instead of `main`.

## Description

<!--
- What is the goal of this pull request? 
- What did you change? 
- Are there any other pull requests, tickets, or issues associated with this pull request?
-->

This adds the password requirements to a central location and refreshes the **Setting a Bootstrap Password** page to use tabs and remove redundant material.

Also: Users are not actually REQUIRED to reset the admin password after first login, though they likely should.

## Comments

<!--
Any additional notes a reviewer should know before we review.
-->

Backticks don't seem to work inside tabs unless they're at the end of a line or on their own line?

A page in the dashboard repo makes some claims about passwords that I'm trying to verify: https://github.com/rancher/dashboard/blob/master/docusaurus/docs/code-base-works/auth-sessions-and-tokens.md

1. ~The minimum password length can be modified by the user (not sure at all if this is relevant to the bootstrap password)~ This doesn't apply to the bootstrap password. Passwords made later must be 12 chars or longer.
2. Password and username must differ

martyav commented:
Re: 

> 1. Can users alter the minimum password length post-installation?
> 2. Is it possible for the user to set up Rancher in such a way that the bootstrap password has a minimum length different from 12 chars?
> 3. Are there additional password requirements, such as the username and password must differ?
>
> https://github.com/rancher/dashboard/blob/master/docusaurus/docs/code-base-works/auth-sessions-and-tokens.md from the dashboard repo claims that users can alter the password length with CATTLE_PASSWORD_MIN_LENGTH, but I'm not sure if that field's exposed to users. It also claims the additional requirement that the username and password must differ, which is quite sensible, but wanted to verify.

![image](https://github.com/rancher/rancher-docs/assets/19174201/dc8944be-7676-47d4-a2fd-062cea3b1d2e)

martyav commented:
@maxsokolovsky Hi, Max. To follow up from this morning, can users alter the minimum length of the admin password (not the special case of the bootstrap -- we have established that it has no min length)? And is there any other validation going on -- for example, requiring that the username and password must differ?

maxsokolovsky commented:
@martyav, the only thing I am seeing in the code and an actual test is that the password must be 12 characters in length.

martyav commented:
- https://github.com/rancher/dashboard/issues/6633 - Parent issue

- https://github.com/rancher/dashboard/pull/6801 - v2.6.9
- https://github.com/rancher/dashboard/pull/6897 - v2.7.2:
  - Prevent users to submit password configuration which is not matching the requirement: min 2 and max 256 chars. 
- https://github.com/rancher/dashboard/pull/7930 - v2.7.2: 
  - Added validation for integers exclusively for min-password changes
  - Changed setting key CATTLE_PASSWORD_MIN_LENGTH to PASSWORD_MIN_LENGTH, to match server returned value
  - Allowed editing settings from environment variables; this is due server configuration have been changed for this case
  - Updated tests - https://github.com/rancher/dashboard/pull/7930/commits/3d5a9810e7dfc2e115e38072f0e045ff3d60bf77

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
+
+<Tabs>
+<TabItem value="Helm">
+Set the following value in the Rancher Helm chart:
```
```suggestion

Set the following value in the Rancher Helm chart:
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
+
+</TabItem>
+<TabItem value="Docker">
+Pass the following value to the Docker install command:
```
```suggestion

Pass the following value to the Docker install command:
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
 
+<Tabs>
+<TabItem value="Helm">
+To retrieve the password, use kubectl:
```
```suggestion

To retrieve the password, use kubectl:
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
```diff
@@ -6,26 +6,66 @@ title: Setting up the Bootstrap Password
+
+</TabItem>
+<TabItem value="Docker">
+To retrieve the password, use the Docker container ID:
```
```suggestion

To retrieve the password, use the Docker container ID:
```

btat commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
````diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
+Set the following value in the Rancher Helm chart:
+
+```yaml
+.Values.bootstrapPassword
````
Line 27 says `run the following commands`, but this isn't a command.

Is this intended to be a snippet of the values.yaml file or a snippet of the Helm install command? In either case, the syntax is incorrect.

btat commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
````diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
+Pass the following value to the Docker install command:
+
+```bash
+-e CATTLE_BOOTSTRAP_PASSWORD=$PASSWORD
````
Is `$PASSWORD` intended to be a Bash variable or just a placeholder?

btat commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
````diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
+To retrieve the password, use the Docker container ID:
+
+```bash
 docker logs  container-id  2>&1 | grep "Bootstrap Password:"
````
```suggestion
docker logs container-id  2>&1 | grep "Bootstrap Password:"
```

martyav commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
+</TabItem>
+<TabItem value="Docker">
+
+To retrieve the password, use the Docker container ID:
```
I'm opting to remove both introductory lines as redundant. I had tried to preserve similar wording as was already on the page, but it isn't really working

martyav commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
```diff
@@ -6,26 +6,70 @@ title: Setting up the Bootstrap Password
 </head>
 
-When Rancher starts for the first time, a password is randomly generated for the first admin user. When the admin first logs in to Rancher, the UI shows commands that can be used to retrieve the bootstrap password. The admin needs to run those commands and log in with the bootstrap password. Then Rancher gives the admin an opportunity to reset the password.
+:::tip
```
Moved and no longer formatted as an info box

btat commented on docs/getting-started/installation-and-upgrade/resources/bootstrap-password.md:
````diff
@@ -6,26 +6,63 @@ title: Setting up the Bootstrap Password
+During [Rancher installation](../install-upgrade-on-a-kubernetes-cluster/install-upgrade-on-a-kubernetes-cluster.md), set `bootstrapPassword` alongside any other flags for the Rancher Helm chart. For example:
+
+```bash
+helm install rancher rancher-<CHART_REPO>/rancher \
````
```suggestion
helm install rancher rancher-<chart-repo>/rancher \
```

Placeholder style consistency.

martyav commented:
Synced with v2.8 to v2.6
